### ISSUE TITLE
[Bug]: crashReporter not tracking renderer crashes without initializing in renderer

### ISSUE THREAD

**Issue by TonyWuu** [electron/electron#30431]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a feature request that matches the one I want to file, without success.

### Electron Version

11.4.11

### What operating system are you using?

macOS

### Operating System Version

macOS Big Sur 11.4

### What arch are you using?

x64

### Last Known Working Electron version

N/A

### Expected Behavior

Without `crashReporter.start()` in the renderer process, crashes at not uploaded to the server. Based on #23265, calling `crashReporter.start()` in the main process only should automatically track crashes occurring in child renderer process.

### Actual Behavior

Renderer crashes are not uploaded without calling `crashReporter.start()` from the renderer process.

### Code Sample
Using `electron-quick-start`
```javascript
// main.js
function createWindow () {
  const crashReporterOption = {
    companyName: 'TestApp',
    submitURL: mySubmitURL,
    productName: 'TestApp',
    autoSubmit: true,
    uploadToServer: true,
    ignoreSystemCrashHandler: false,
    compress: false,
  };
  crashReporter.start(crashReporterOption);
... 
}
```

```javascript
// preload.js
...
const {crashReporter} = require('electron')
const crashReporterOption = {
    companyName: 'TestApp',
    submitURL: mySubmitURL,
    productName: 'TestApp',
    autoSubmit: true,
    uploadToServer: true,
    ignoreSystemCrashHandler: false,
    compress: false,
};

crashReporter.start(crashReporterOption); // omitting this line causes results in renderer crashes not being uploaded; crashes still appear in Crashpad directory
process.once('loaded', () => {
  const _process = process;
  global.process = {};
  global.process.crash = _process.crash;
});
```

```javascript
// renderer.js
process.crash();
```

### Additional Information

Is there a way I can further debug why the upload is not occurring/failing?

**Comment by clavin** [comment]
Hi! Thanks for reporting this issue :)

Could you clarify how this is broken in 9.4.3 yet is also "last known working" in 9.4.3? Do you mean that this functionality somehow broke without any update to Electron? Or do you mean something else?

It would also be a big help if you could make a small test case that a maintainer can run to quickly reproduce this bug and debug it themselves, alongside some steps to trigger the bug! I'm going to add the `blocked/need-repro` to this issue until there is one. Please feel free to link to it in a followup comment.

Thanks in advance!

**Comment by miniak** [comment]
@TonyWuu Electron 9 is not supported, you need to reproduce this on at least Electron 11 for this issue to be actionable.

**Comment by TonyWuu** [comment]
@clavin @miniak thanks, I've updated the original post with further details and a minimal repro case

**Comment by TonyWuu** [comment]
Was also able to repro on Electron 11

**Comment by TonyWuu** [comment]
@nornagon any additional insight you could provide here? Given that no one else has reported the issue, makes me thing it's just me doing something wrong..

**Comment by nornagon** [comment]
@TonyWuu that's very weird, and we do not see the same behavior.

Are the crashes not being captured at all, or just not uploaded?

I see you have `ignoreSystemCrashHandler: false`. Sometimes macOS's crash reporter can take a very long time to process a crash. Does the issue still occur if you elide that option?

**Comment by TonyWuu** [comment]
> Are the crashes not being captured at all, or just not uploaded?

@nornagon The crashes are being captured every time in Application Support; they're just not uploaded. Is there a way to debug why they're not being uploaded or failing to upload?

> I see you have `ignoreSystemCrashHandler: false`. Sometimes macOS's crash reporter can take a very long time to process a crash. Does the issue still occur if you elide that option?

Setting `ignoreSystemCrashHandler: true` didn't help

It's worth nothing that once I include `crashReporter.start(crashReporterOption);` in the preload script, the renderer crashes are correctly being captured again

**Comment by nornagon** [comment]
@TonyWuu ah, I think the problem is that you are using an old version of Electron. The behavior of `crashReporter` changed in Electron 12: https://www.electronjs.org/docs/breaking-changes#removed-crashreporter-methods-in-the-renderer-process so the documentation is not accurate for Electron 11.

I'm closing this as Electron 11 is [no longer supported](https://www.electronjs.org/docs/tutorial/support#currently-supported-versions).

**Comment by TonyWuu** [comment]
Sorry to drag on a closed thread, but I was still able to reproduce this in Electron 13. It seems however that calling `crashReporter.start` in the preload script still works and isn't deprecated, and fixes the issue to track renderer crashes, so will just continue doing that for now :/

**Comment by nornagon** [comment]
That's surprising, because `crashReporter.start` doesn't exist in 13.x: https://github.com/electron/electron/blob/13-x-y/lib/renderer/api/crash-reporter.ts

**Comment by TonyWuu** [comment]
@nornagon ugh you're absolutely right, I forgot to run npm install after bumping the version... 🤦‍♂️  the error correctly shows now
![image](https://user-images.githubusercontent.com/16580702/132590445-7b201efd-9b04-4889-859d-5980c2b3917f.png)

Anyhow, I'm still not getting renderer process crashes on the server. Main process crashes are appearing correctly. My minimum repro case:

``` javascript
// main.js
function createWindow () {
  const crashReporterOption = {
    companyName: 'TestApp',
    submitURL: mySubmitURL,
    productName: 'TestApp',
    autoSubmit: true,
    uploadToServer: true,
    ignoreSystemCrashHandler: false,
    compress: false,
  };
  crashReporter.start(crashReporterOption);

  // Create the browser window.
  const mainWindow = new BrowserWindow({
    width: 800,
    height: 600,
    webPreferences: {
      preload: path.join(__dirname, 'preload.js'),
      worldSafeExecuteJavascript: false,
      contextIsolation: false,
    }
  })
  ...
}
```

``` javascript
// preload.js
...
process.once('loaded', () => {
  const _process = process;
  global.process = {};
  global.process.crash = _process.crash;
});
```

```javascript
// renderer.js
process.crash()
```

As before, main process and renderer process crashes are both appearing in the Crashpad folder, but only main process crashes uploaded for some reason. Testing on v12.1.0

**Comment by TonyWuu** [comment]
Was able to figure out what the issue was. We were passing an unexpected user agent string in the 'extra' parameters to the crash reporting server from the renderer process. 
This wasn't an issue before because we used to start the crash reporter from the renderer with this info in the extra start options; now we just need to explicitly add the extra parameters by calling `crashReporter.addExtraParameter()`

Closing!
